src/app/static/src/js/types.ts

```diff
@@ -0,0 +1,19 @@
+export interface Error {
+    response?: {
+        data?: {
+            errors?: any[]
+        }
+    }
+}
+
+export interface ReportDependency {
+    id: string,
+    name: string,
+    out_of_date: boolean,
+    dependencies: ReportDependency[]
+}
+
+export interface ReportDependencies {
+    direction: string, //upstream or downstream
```

Comment:
```suggestion
    direction: "upstream" | "downstream"
```

Also: we have already have [utils/types.ts](https://github.com/vimc/orderly-web/blob/master/src/app/static/src/js/utils/types.ts). Should they be combined?